### PR TITLE
merge realistic digitization with Ruben's development 

### PR DESCRIPTION
**Pull request by AllaMaevskaya**

### CODE REVIEW

File: Detectors/FIT/common/simulation/include/FITSimulation/HitType.h [hunk 1]
```diff
@@ -19,7 +19,6 @@ class HitType : public o2::BasicXYZEHit<float>
 
 } // namespace o2
```
Comment (shahor02) [inline Detectors/FIT/common/simulation/include/FITSimulation/HitType.h]: It looks like you did not rebase on dev, the license has disappeared..

File: Common/Field/include/Field/MagneticWrapperChebyshev.h [hunk 2]
```diff
@@ -180,24 +180,24 @@ class MagneticWrapperChebyshev : public TNamed
       return mMaxRadiusTPCRat;
```
Comment (shahor02) [inline Common/Field/include/Field/MagneticWrapperChebyshev.h]: Here a lot of files which don't belong to this PR are modified...

File: Detectors/FIT/common/simulation/src/Digitizer.cxx [hunk 3]
```diff
@@ -24,15 +25,25 @@ using namespace o2::fit;
...
+  double maximum{ hist.GetMaximum() };
+  int binfound{ hist.FindFirstBinAbove(maximum * 0.2) };
+  //  std::cout<<"@@@@@!!! max "<<maximum<<" binfound "<<binfound<<" return "<<hist.GetBinCenter(binfound)<<std::endl;
+  return hist.GetBinCenter(binfound);
+}
 
-  constexpr Float_t C_side_cable_cmps = 2.877; //ns
-  constexpr Float_t A_side_cable_cmps = 11.08; //ns
-  constexpr Float_t signal_width = 5.;         // time gate for signal, ns
+void Digitizer::process(const std::vector<o2::fit::HitType>* hits, Digit* digit, std::vector<std::vector<double>>& channel_times)
```
Comment (shahor02) [inline Detectors/FIT/common/simulation/src/Digitizer.cxx]: Do you really need vector of vectors? It seems in the process method you always set the outer vector to the same size (mMCP). If this is fixed, better use array<vectot<double>,size).

Comment (shahor02) [reply Detectors/FIT/common/simulation/src/Digitizer.cxx]: Hi @AllaMaevskaya 

I think the best would be to unstage your changes (``git reset --soft HEAD~7``), then recover all files which should not be affected by this PR (``git checkout -- Common/Field/include/Field/MagneticWrapperChebyshev.h`` etc.), then rebase to fresh dev, solve conflicts if any and push again.

Cheers,
 Ruben

Comment (AllaMaevskaya) [reply Detectors/FIT/common/simulation/src/Digitizer.cxx]: Hello
mMCP=208 is the number of channels for T0 but it can be different for V0. So better to keep it as it is before we will decide how to call it for both detectors

Best regards
Alla

File: Steer/DigitizerWorkflow/src/FITDigitizerSpec.cxx [hunk 4]
```diff
@@ -207,6 +210,7 @@ o2::framework::DataProcessorSpec getFITT0DigitizerSpec(int channel)
              { "simFileS", VariantType::String, "", { "Sim (signal) input filename" } },
              { "pileup", VariantType::Int, 1, { "whether to run in continuous time mode" } } }
```
Comment (shahor02) [inline Steer/DigitizerWorkflow/src/FITDigitizerSpec.cxx]: See e.g. https://github.com/AliceO2Group/AliceO2/blob/d99d43997bfc2377bc52d738856894cf4af8d4f9/Steer/DigitizerWorkflow/src/TPCDigitizerSpec.cxx#L350 on using VariantType::Bool in similar situation. BTW, does FIT have triggered mode?

File: Detectors/FIT/common/simulation/src/Digitizer.cxx (hunk 3 above)
Comment (shahor02) [reply Detectors/FIT/common/simulation/src/Digitizer.cxx]: @AllaMaevskaya ok, I still did not understand if the Digitizer will be serving only T0 or all subdetectors.
